src/wovnio/html/HtmlConverter.php

```diff
@@ -45,98 +44,89 @@ public function __construct($html, $encoding, $token, $store = null, $headers =
-    $this->insertSnippet($dom);
-    if (isset($this->store) && isset($this->headers)) {
-      $this->insertHreflangTags($dom);
-    }
-
-    $marker = new HtmlReplaceMarker();
-    if ($removeParts) {
-      $this->removeWovnIgnore($dom, $marker);
-      $this->removeForm($dom, $marker);
-      $this->removeScript($dom, $marker);
-    }
-
-    $converted_html = $dom->save();
-
-    // Without clear(), Segmentation fault will be raised.
-    // @see https://sourceforge.net/p/simplehtmldom/bugs/103/
-    $dom->clear();
-    unset($dom);
+    $this->html = $this->insertSnippet($this->html);
+    $this->html = $this->insertHreflangTags($this->html);
 
-    if ($removeParts) {
-      $converted_html = $this->removeBackendWovnIgnoreComment($converted_html, $marker);
-    }
-    return array($converted_html, $marker);
+    return array($this->html);
   }
 
   /**
    * Insert wovn's snippet to ensure snippet is always inserted.
    * When snippet is always inserted, do nothing
    *
-   * @param simple_html_dom_node $dom
+   * @param string $html
    */
-  private function insertSnippet($dom)
+  private function insertSnippet($html)
   {
-    foreach ($dom->find('script') as $node) {
-      if (strpos($node->getAttribute('src'), '//j.wovn.io/') !== false ||
-        strpos($node->getAttribute('src'), '//j.dev-wovn.io:3000/') !== false) {
-        return;
-      }
+    $snippet_regex = "<\<script(.*)src='(j.wovn.io|j.dev-wovn.io)'(.*)><\/script>";
+    if (preg_match("/$snippet_regex/i", $html)) {
+      return;
     }
 
-    $token = $this->token;
-    $insert_tags = array('head', 'body', 'html');
-    foreach ($insert_tags as $tag_name) {
-      $parents = $dom->find($tag_name);
-      if (count($parents) > 0) {
-        $parent = $parents[0];
-        $parent->innertext = "<script src='//j.wovn.io/1' data-wovnio='key=$token' data-wovnio-type='backend_without_api' async></script>" . $parent->innertext;
-        return;
+    $snippet_code = $this->buildSnippetCode();
+    $parent_tags = array("(<head\s?.*?>)", "(<body\s?.*?>)", "(<html\s?.*?>)");
+
+    return $this->insertAfterTag($parent_tags, $html, $snippet_code);
+  }
+
+  private function insertAfterTag($tag_names, $html, $insert_str)
+  {
+    foreach ($tag_names as $tag_name) {
+      if (preg_match($tag_name, $html, $matches, PREG_OFFSET_CAPTURE)) {
+        return substr_replace($html, $insert_str, $matches[0][1] + strlen($matches[0][0]), 0);
       }
     }
   }
 
+  private function buildSnippetCode()
+  {
+    $token = $this->token;
+    $current_lang = $this->headers->lang();
+    $default_lang = $this->store->settings['default_lang'];
+    $url_pattern = $this->store->settings['url_pattern_name'];
+    $lang_code_aliases_json = json_encode($this->store->settings['custom_lang_aliases']);
+//    $snippet_code = "<script src='//j.wovn.io/1' data-wovnio='key=$token' data-wovnio-type='backend_without_api' async></script>";
```

Comment:
This is the old incorrect one, so we can remove this comment now, right?

Reply:
Yes. I will remove

Reply:
Removed. Thanks